crates/services/gas_price_service/src/common/fuel_core_storage_adapter.rs

```diff
@@ -43,17 +52,41 @@ use fuel_core_types::{
         Transaction,
     },
 };
+use fuel_gas_price_algorithm::v1::{
+    Bytes,
+    Error,
+    Height,
+    UnrecordedBlocks,
+};
 use std::cmp::min;
 
 #[cfg(test)]
 mod metadata_tests;
 
 pub mod storage;
 
-impl<Storage> MetadataStorage for StructuredStorage<Storage>
+impl<Storage> SetMetadataStorage for StructuredStorage<Storage>
 where
     Storage: KeyValueInspect<Column = GasPriceColumn> + Modifiable,
     Storage: Send + Sync,
+{
+    fn set_metadata(&mut self, metadata: &UpdaterMetadata) -> GasPriceResult<()> {
+        let block_height = metadata.l2_block_height();
+        let mut tx = self.write_transaction();
+        tx.storage_as_mut::<GasPriceMetadata>()
```

Comment:
Why are we creating a tx for 1 insert ?

Reply:
I feel like that's what we do most places.

Reply:
If we constrained with `KeyValueMutate` we could use `put` directly, but that's not what we have access to.